### ISSUE TITLE
Can't build Marktext

### ISSUE THREAD

**Issue by Leoriem-code** [marktext/marktext#2839]
### Description

When installing yarn to build Marktext, the build fail because of `fsevents`, `dmg-license` and `iconv-corefoundation`, always with this error: `The linux-x64 architecture is incompatible with this module, link skipped.`

- [x] Can you reproduce the issue? Yes I can

### Steps to reproduce
1. ```bash
   $ sudo npm install -g yarn
   changed 1 package, and audited 2 packages in 3s
   found 0 vulnerabilities
   ```

2. ```bash
   $ yarn -v
   1.22.17
   ```

3. ```bash
   $ yarn set version stable
   ➤ YN0000: Retrieving https://repo.yarnpkg.com/3.1.1/packages/yarnpkg-cli/bin/yarn.js
   ➤ YN0000: Saving the new release in .yarn/releases/yarn-3.1.1.cjs
   ➤ YN0000: Done in 1s 520ms
   ```

4. ```bash
   $ yarn -v
   3.1.1
   ```

5. ```bash
   $ yarn install
   ➤ YN0070: Migrating from Yarn 1; automatically enabling the compatibility node-modules linker :thumbs_up:
   
   ➤ YN0000: ┌ Resolution step
   ➤ YN0032: │ ced@npm:2.0.0: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ fontmanager-redux@npm:1.1.0: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ keyboard-layout@npm:2.0.17: Implicit dependencies on node-gyp are discouraged
   ➤ YN0061: │ popper.js@npm:1.16.1 is deprecated: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
   ➤ YN0032: │ keytar@npm:7.7.0: Implicit dependencies on node-gyp are discouraged
   ➤ YN0061: │ debug@npm:3.2.6 is deprecated: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
   ➤ YN0061: │ debug@npm:4.1.1 is deprecated: Debug versions >=3.2.0 <3.2.7 || >=4 <4.3.1 have a low-severity ReDos regression when used in a Node.js environment. It is recommended you upgrade to 3.2.7 or 4.3.1. (https://github.com/visionmedia/debug/issues/797)
   ➤ YN0013: │ file-icons@https://github.com/file-icons/atom.git#commit=66d57920f352fa5ed0c66f6fe86a98b2b3b8dac1 can't be found in the cache and
   ➤ YN0032: │ lzma-native@npm:8.0.1: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ fsevents@npm:2.3.2: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ node-addon-api@npm:3.0.2: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ node-addon-api@npm:3.2.1: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ nan@npm:2.14.1: Implicit dependencies on node-gyp are discouraged
   ➤ YN0061: │ highlight.js@npm:9.18.1 is deprecated: Version no longer supported. Upgrade to @latest
   ➤ YN0061: │ querystring@npm:0.2.0 is deprecated: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
   ➤ YN0032: │ cld@npm:2.7.1: Implicit dependencies on node-gyp are discouraged
   ➤ YN0032: │ @hfelix/spellchecker@npm:4.1.0: Implicit dependencies on node-gyp are discouraged
   ➤ YN0061: │ @types/fast-json-stable-stringify@npm:2.1.0 is deprecated: This is a stub types definition. fast-json-stable-stringify provides its own type definitions, so you do not need this installed.
   ➤ YN0061: │ uuid@npm:3.4.0 is deprecated: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
   ➤ YN0061: │ core-js@npm:2.6.11 is deprecated: core-js@<3.4 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Please, upgrade your dependencies to the actual version of core-js.
   ➤ YN0061: │ resolve-url@npm:0.2.1 is deprecated: https://github.com/lydell/resolve-url#deprecated
   ➤ YN0061: │ urix@npm:0.1.0 is deprecated: Please see https://github.com/lydell/urix#deprecated
   ➤ YN0032: │ node-addon-api@npm:1.7.2: Implicit dependencies on node-gyp are discouraged
   ➤ YN0061: │ xmldom@npm:0.1.31 is deprecated: Deprecated due to CVE-2021-21366 resolved in 0.5.0
   ➤ YN0061: │ ini@npm:1.3.5 is deprecated: Please update to ini >=1.3.6 to avoid a prototype pollution issue
   ➤ YN0002: │ marktext@workspace:. doesn't provide @vue/compiler-sfc (p7738d), requested by vue-loader
   ➤ YN0060: │ marktext@workspace:. provides eslint (p231b2) with version 8.5.0, which doesn't satisfy what eslint-config-standard requests
   ➤ YN0060: │ marktext@workspace:. provides eslint-plugin-promise (pae6f3) with version 6.0.0, which doesn't satisfy what eslint-config-standard requests
   ➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
   ➤ YN0000: └ Completed in 23s 716ms
   ➤ YN0000: ┌ Fetch step
   ➤ YN0013: | [All dependencies show: "can't be found in the cache and will be fetched from the remote registry"]
   ➤ YN0000: └ Completed in 16s 80ms
   ```  

On a quick side-note, because of those message:
```bash
➤ YN0002: │ marktext@workspace:. doesn't provide @vue/compiler-sfc (p7738d), requested by vue-loader
➤ YN0060: │ marktext@workspace:. provides eslint (p231b2) with version 8.5.0, which doesn't satisfy what eslint-config-standard requests
➤ YN0060: │ marktext@workspace:. provides eslint-plugin-promise (pae6f3) with version 6.0.0, which doesn't satisfy what eslint-config-standard requests
```
I had to change [`packages.json`](https://github.com/marktext/marktext/blob/develop/package.json) to remove these errors (but it's not marktext errors, so not included)
```
'"@vue/compiler-sfc": "^3.2.26",' (New)'
"eslint": "^8.5.0",' -> '"eslint": "^7.12.1",'
'"eslint-plugin-promise": "^6.0.0",' -> '"eslint-plugin-promise": "^4.2.1 || ^5.0.0",'
```  

**Expected behavior:**
The build would complete without failure

**Actual behavior:**
```bash
➤ YN0000: ┌ Link step
➤ YN0076: │ fsevents@patch:fsevents@npm%3A2.3.2#~builtin<compat/fsevents>::version=2.3.2&hash=18f3a7 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0076: │ dmg-license@npm:1.0.9 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0076: │ iconv-corefoundation@npm:1.1.6 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0007: │ marktext@workspace:. must be built because it never has been before or the last one failed
➤ YN0009: │ marktext@workspace:. couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-dbd74c29/build.log)
➤ YN0000: └ Completed in 7s 273ms
```
The log isn't useful at all, because all he said is `[ERROR] Please use yarn to install dependencies.`

### Versions

- MarkText version: 0.16.3
- Operating system: Debian-based

**Comment by fxha** [comment]
> Migrating from Yarn 1; automatically enabling the compatibility

That seems wrong to me. Please use yarn v1 and try to uncomment https://github.com/marktext/marktext/blob/f3caf90920cc2fa312f8191d5ae3f7c0c62a556b/.electron-vue/preinstall.js#L9 if there are problems with `Please use yarn to install dependencies`.

**Comment by Leoriem-code** [comment]
So, thank you @fxha, I managed to build it.
I had to comment lines 9 to 12 of `preinstall.js` (otherwise the error showed up) and I did run `yarn install` and `yarn run build` successfully.
I choose to run again `yarn set version stable` and `yarn install` because I wanted to see if anything changed and *surprise surprise!* yarn did ignore `fsevents`, `dmg-license` and `iconv-corefoundation` (which he didn't before) so I managed to run everything smoothly.
In short, if anyone have the same problem, try this, it may work.
